go/engine/pgp_purge.go

```diff
@@ -61,80 +61,76 @@ func (e *PGPPurge) Run(ctx *Context) error {
 	}
 	e.me = me
 
-	for _, bundle := range e.me.GetActivePGPKeys(false) {
-		if err := e.export(ctx, bundle); err != nil {
-			return err
-		}
+	// get all PGP blocks in keyring
+	var blocks []*libkb.SKB
+	kerr := e.G().LoginState().Keyring(func(ring *libkb.SKBKeyringFile) {
+		blocks, err = ring.AllPGPBlocks()
+	}, "AllPGPBlocks")
+	if kerr != nil {
+		return kerr
 	}
-
-	return nil
-}
-
-// KeyFiles returns the filenames of the exported keys.
-func (e *PGPPurge) KeyFiles() []string {
-	return e.filenames
-}
-
-func (e *PGPPurge) export(ctx *Context, bundle *libkb.PGPKeyBundle) error {
-	skb, key, err := e.findAndUnlock(ctx, bundle)
 	if err != nil {
 		return err
 	}
-	if key == nil {
-		return nil
-	}
 
-	filename := filepath.Join(e.G().Env.GetConfigDir(), key.GetFingerprint().String()+".sp")
-	if err := e.encryptToFile(ctx, key, filename); err != nil {
+	// export each one to a file
+	if err := e.exportBlocks(ctx, blocks); err != nil {
 		return err
 	}
 
 	if e.arg.DoPurge {
-		e.G().Log.Debug("Removing PGP Key %s from keyring", key.GetFingerprint())
-		if err := libkb.RemoveSKBFromKeyring(e.G(), skb); err != nil {
+		// if purge flag set, remove all PGP blocks from keyring and save it
+		kerr = e.G().LoginState().Keyring(func(ring *libkb.SKBKeyringFile) {
+			err = ring.RemoveAllPGPBlocks()
+			if err == nil {
+				err = ring.Save()
+			}
+		}, "RemoveAllPGPBlocks")
+		if kerr != nil {
+			return kerr
+		}
+		if err != nil {
 			return err
 		}
 	}
 
-	e.filenames = append(e.filenames, filename)
 	return nil
 }
 
-func (e *PGPPurge) findAndUnlock(ctx *Context, bundle *libkb.PGPKeyBundle) (*libkb.SKB, *libkb.PGPKeyBundle, error) {
-	arg := libkb.SecretKeyArg{
-		Me:       e.me,
-		KeyType:  libkb.PGPKeyType,
-		KeyQuery: bundle.GetFingerprint().String(),
-	}
-	var skb *libkb.SKB
-	var err error
-	aerr := e.G().LoginState().Account(func(a *libkb.Account) {
-		skb, err = a.LockedLocalSecretKey(arg)
-	}, "PGPPurge - export")
-	if aerr != nil {
-		return nil, nil, aerr
-	}
-	if err != nil {
-		return nil, nil, err
-	}
-	if skb == nil {
-		return nil, nil, nil
-	}
+// KeyFiles returns the filenames of the exported keys.
+func (e *PGPPurge) KeyFiles() []string {
+	return e.filenames
+}
 
-	secretRetriever := libkb.NewSecretStore(e.G(), e.me.GetNormalizedName())
+func (e *PGPPurge) exportBlocks(ctx *Context, blocks []*libkb.SKB) error {
+	sstore := libkb.NewSecretStore(e.G(), e.me.GetNormalizedName())
 	promptArg := libkb.SecretKeyPromptArg{
 		SecretUI: ctx.SecretUI,
 		Reason:   "export private PGP key",
 	}
-	gk, err := skb.PromptAndUnlock(promptArg, secretRetriever, e.me)
-	if err != nil {
-		return nil, nil, err
-	}
-	pk, ok := gk.(*libkb.PGPKeyBundle)
-	if !ok {
-		return nil, nil, fmt.Errorf("unlocked key incorrect type")
+
+	for i, block := range blocks {
+		block.SetUID(e.me.GetUID())
+		key, err := block.PromptAndUnlock(promptArg, sstore, e.me)
```

Comment:
you shouldn't get prompted at all, but if you do, it should only happen once after secret store PR gets in.